content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -117,12 +121,74 @@ kind: KubeletConfiguration
 address: "192.168.0.8"
 ```
 
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+evictionHard:
+  memory.available: "200Mi"
+address: "192.168.0.8"
+```
+Additionally, you can change the slices/lists values of the kubelet configuration. For example, you can override the `clusterDNS` and `featureGates` fields as follows:
+
+Main kubelet configuration file contents:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+evictionHard:
+  memory.available: "200Mi"
+clusterDNS:
+  - "192.168.0.9"
+  - "192.168.0.8"
+featureGates:
+  MemoryQoS: true
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+clusterDNS:
```

Comment:
how eviction collections will be merged? Will the whole structure be overriden or one value added?

Reply:
Same question about logging - will the whole structure be overriden or individual fields?

Reply:
The merging logic is targeted for the individual fields. If the intention is to override fields, then mention only those fields in the drop-in configs.

Reply:
individual high level fields or all of them? I simply don't know how it works and this documentation doesn't give a clear understanding.

Reply:
evictionHard and evictionSoft is coming from - what if I want to configur MemorySwap. I need to enable feature gate, and potentially want to disable eviction hard and evictionSoft. Is it possible to do with a single drop-in file?

Reply:
>evictionHard and evictionSoft is coming from - what if I want to configur MemorySwap. I need to enable feature gate, and potentially want to disable eviction hard and evictionSoft. Is it possible to do with a single drop-in file?

Yes, you can set the values in a drop-in config for this use case.

Reply:
>individual high level fields or all of them? I simply don't know how it works and this documentation doesn't give a clear understanding.

Please scratch the previous response.

>how eviction collections will be merged? Will the whole structure be overriden or one value added?

In this case, the entire structure will be overridden. If you want to change just one field, you can certainly do that. For example, in the case of `evictionHard`, you can alter the `memory.available` to a new value. However, you have to set values for other fields, perhaps using default values or any other values you desire, to make that happen.